dulwich/tests/test_client.py

```diff
@@ -984,6 +984,22 @@ def test_from_parsedurl_on_url_with_quoted_credentials(self):
         expected_basic_auth = 'Basic %s' % str(b64_credentials)
         self.assertEqual(basic_auth, expected_basic_auth)
 
+    def test_url_redirect_location(self):
```

Comment:
This test depends on internet access, which means that if e.g. one of these external sites changes, the test breaks. Internet access is also not available e.g. as part of Debian package builds. Could you change it to avoid the network access, e.g. by mocking?

Reply:
Sure, I will try to work something out. I will ping you when it is done.